global_helpers/panther_base_helpers.py

```diff
@@ -309,6 +310,60 @@ def deep_get(dictionary: dict, *keys, default=None):
     return out
 
 
+# pylint: disable=R1260,R0911
+def deep_walk(obj: dict, *keys: str, default: str = None, return_val: str = "all") -> Optional[str]:
+    """Safely retrieve a value stored in complex dictionary structure
```

Comment:
I love this explainer, and I would love it more if this were formatted like a pydoc ala [LookupTableHelpers.p_match](https://github.com/panther-labs/panther-analysis/blob/master/global_helpers/panther_lookuptable_helpers.py#L27-L34)

Reply:
Will do!

Reply:
Addressed in [`37575e8` (#819)](https://github.com/panther-labs/panther-analysis/pull/819/commits/37575e805c2c8a2cb3b7ad02c92bc00bc693fa42).